sys-apps/plocate/plocate-1.1.5.ebuild

```diff
@@ -0,0 +1,54 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit linux-info meson systemd
+
+DESCRIPTION="Posting locate is a much faster locate"
+HOMEPAGE="https://plocate.sesse.net/"
+SRC_URI="https://plocate.sesse.net/download/${P}.tar.gz"
+
+LICENSE="GPL-2 GPL-2+"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE="io-uring"
+
+RDEPEND="
+	acct-group/locate
+	app-arch/zstd:=
+	io-uring? ( sys-libs/liburing:= )
+	!sys-apps/mlocate
+"
+DEPEND="${RDEPEND}"
+
+CONFIG_CHECK="~IO_URING"
+ERROR_IO_URING="required for USE=io-uring"
+
+PATCHES=(
+	"${FILESDIR}/${PN}"-1.1.5-no-systemd-check.patch
+)
+
+src_prepare() {
+	# pretend liburing dep doesn't exist when USE flag off
+	if ! use io-uring; then
+		sed -i "s/dependency('liburing/dependency('/" meson.build || die
+	fi
+
+	default
+}
+
+src_configure() {
+	local emesonargs=(
+		-Dlocategroup=locate
+		-Dinstall_cron=true
```

Comment:
this is misleading, out of the box the cron script plocate installs is not very useful, only converting an mlocate-style database into a plocate style one:

```sh
#! /bin/bash

set -e

if [ @PROCESSED_BY_MESON@ = 1 ]; then
        SBINDIR=@sbindir@
        LOCATEGROUP=@groupname@
        DBFILE=@dbfile@
else
        SBINDIR=/usr/local/sbin
        LOCATEGROUP=plocate
        DBFILE=/var/lib/plocate/plocate.db
fi

$SBINDIR/plocate-build /var/lib/mlocate/mlocate.db $DBFILE.new
chgrp $LOCATEGROUP $DBFILE.new
mv $DBFILE.new $DBFILE
```

see: https://gitweb.gentoo.org/repo/proj/guru.git/commit/?id=2aaa3ab6406398c4f531e32b0284dbc7d5c6b5d3

perhaps the cron script in there should utilize is_ac_power or similar, to bring it up to main repo standards

Reply:
Thanks for pointing that out. I've added most of the files from mlocate's FILESDIR for feature parity, including the default updatedb.conf, and cron script.